lasagne/layers/cuda_convnet.py

```diff
@@ -29,17 +29,177 @@
     raise ImportError("requires a GPU to work")
 
 
-# TODO: make sure to document the limitations and 'best practices'
-# (i.e. minibatch size % 128 == 0)
-# TODO: see if the 'dimshuffle' logic can be put in the base class instead.
-
-
 # base class for all layers that use ops from pylearn2.sandbox.cuda_convnet
 class CCLayer(Layer):
     pass
 
 
 class Conv2DCCLayer(CCLayer):
+    """
+    2D convolutional layer
+
+    Performs a 2D convolution on its input and optionally adds a bias and
+    applies an elementwise nonlinearity.  This is an alternative implementation
+    which uses the cuda-convnet wrappers from pylearn2:
+    ``pylearn2.sandbox.cuda_convnet.filter_acts.FilterActs``.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 4D tensor, with shape
+        ``(batch_size, num_input_channels, input_height, input_width)``.
+        If automatic dimshuffling is disabled (see notes), the shape should be
+        ``(num_input_channels, input_height, input_width, batch_size)``
```

Comment:
Elsewhere we settled for `input_rows` and `input_columns` to not take a position on what's height and what's width.